### PR TITLE
Prevent sys.prefix from leaking into child process on macOS

### PR DESCRIPTION
**Pull request by cjolowicz**
Fixes https://github.com/pypa/virtualenv/issues/1643

---

## Thanks for contributing a pull request, see checklist all is good!

- [ ] wrote descriptive pull request text
- [ ] added/updated test(s)
- [ ] updated/extended the documentation
- [ ] added news fragment in ``docs/changelog`` folder

### CODE REVIEW

File: src/virtualenv/discovery/cached_py_info.py [hunk 1]
```diff
@@ -103,6 +103,10 @@ def _run_subprocess(cls, exe):
     with ensure_file_on_disk(resolved_path) as resolved_path:
         cmd = [exe, "-s", str(resolved_path)]
 
+        # Prevent sys.prefix from leaking into the child process on macOS.
+        if sys.platform == "darwin":
```
Comment (gaborbernat) [inline src/virtualenv/discovery/cached_py_info.py]: I have a feeling that we should do this unconditionally 🤔 that being said I'm not a fan of modifying the os.environ directly; instead copy it, modify it and then pass it into the Popen

Comment (gaborbernat) [reply src/virtualenv/discovery/cached_py_info.py]: we should also add some test; and changelog for this

Comment (gaborbernat) [reply src/virtualenv/discovery/cached_py_info.py]: oh and do link the upstream bug ticket as a comment 👍

**Comment by cjolowicz** [comment]
Thanks for merging! ...and sorry, did not get around to writing test and changelog earlier. 

Updated PR description with repro and analysis.

**Pull request merged**